### PR TITLE
Include OpenBazaar Server

### PR DESCRIPTION
**Pull request by swordfish6975**
Once diet-pi has rebooted:

```
cd OpenBazaar-Server 
nano ob.cfg  
```

Set  the username and password - remember to remove the #
ctrl x and y to save

`python openbazaard.py start -d -a [IP of machine allowed to connect or 0.0.0.0]`

Note: if you want to see the log omit -d

### CODE REVIEW

File: dietpi/dietpi-software [hunk 1]
```diff
@@ -2906,6 +2941,14 @@ _EOF_
 			WEBSERVER_REDIS=2
 		fi
 
+		
+		#OPENBAZAAR
+		if (( $OPENBAZAAR == 1 )); then
+	
+			OPENBAZAAR=2
```
Comment (swordfish6975) [inline dietpi/dietpi-software:2948]: not sure if I need this if statement

Comment (Fourdee) [inline dietpi/dietpi-software:2948]: Yep it is. Sets the program to `2` which is "Installed"

File: dietpi/dietpi-software [hunk 2]
```diff
@@ -721,6 +722,40 @@ INDEX_WEBSERVER_TARGET=${INDEX_WEBSERVER_TARGET:=-2}
...
 		fi
+		
+		#OPENBAZAAR
+		if (( $OPENBAZAAR == 1 )); then
+		
+			INSTALL_DESCRIPTION='OpenBazaar'
+			Banner_Installing
+		    
+			#apt-get -y update && apt-get -y upgrade
+			apt-get -y install git
```
Comment (Fourdee) [inline dietpi/dietpi-software]: Use `AGI` function to `apt-get install -y`. AGI has error checks and some other stuff, see function `AGI()` for more details.

File: dietpi/dietpi-software [hunk 3]
```diff
@@ -721,6 +722,40 @@ INDEX_WEBSERVER_TARGET=${INDEX_WEBSERVER_TARGET:=-2}
...
+			apt-get -y install build-essential libssl-dev libffi-dev python-dev openssl python-pip libzmq3-dev
```
Comment (Fourdee) [inline dietpi/dietpi-software]: `build-essential` and `git` should trigger the DietPi installations of them in this script 

Add entry for `OPENBAZAAR` to trigger these.

```
        #DietPi software that requires GITCLIENT
        if (( $EMONCMS == 1 )); then
            GITCLIENT=1
        elif (( $AMPACHE == 1 )); then
            GITCLIENT=1
        fi

        #DietPi software that requires Build-Essential
        if (( $HAPROXY == 1 )); then
            BUILDESSENTIAL=1
        elif (( $WIRINGPI == 1 )); then
            BUILDESSENTIAL=1
        fi

```

**Comment by Fourdee** [comment]
@swordfish6975 
Excellent work on this :+1: 

I've made a few comments, I can either merge this and I will make the modifications, or if you are up for it, you are more than welcome to make the changes on this pull request (to give you full credit).

**Comment by mariodian** [comment]
Have you tested it? I've installed OB on many linux machines and I always had to run `pip install cryptography` and `make install` with sudo, otherwise OB server wouldn't start.

**Comment by swordfish6975** [comment]
Yep tested on my Orange Pi PC, my store @swordfish6975 is running from there now

**Comment by swordfish6975** [comment]
I do get this though, but it still seems to work.

```
root@DietPi:~/OpenBazaar-Server# python openbazaard.py start -d -a 0.0.0.0
/usr/local/lib/python2.7/dist-packages/cffi-1.6.0-py2.7-linux-armv7l.egg/cffi/model.py:526: UserWarning:
'point_conversion_form_t' has no values explicitly defined; next version will refuse to guess which
integer type it is meant to be (unsigned/signed, int/long)
  % self._get_c_name())

```

Edit: Perhaps related to [this](https://github.com/pyca/pyopenssl/issues/392)

**Comment by Fourdee** [comment]
@swordfish6975 
Thanks for updating the code, good job :+1: 

I'll mark this for v118 so we can get it in for next release. I'am out all day today, so I will merge this most likely tomorrow then run some tests straight after.

I noticed you tested with OPi PC. I'll make a note to test the other devices to cover all architectures.

> note to self test:
> C2
> RPi Jessie
> VM

**Comment by swordfish6975** [comment]
I did most dev testing in VirtualBox as it was faster and less fiddling with SD cards.

I have tested 0906609 with OPi PC  - > this is the SD card I am running my store on now.

Also tested 4cf23d0 with OPi PC and VirtualBox and wrote [these instructions](https://www.reddit.com/r/OpenBazaar/comments/4i97my/when_this_gets_merged_and_released_it_will_make/d2wphr5)

**Comment by Fourdee** [comment]
@swordfish6975 
Excellent :+1: 
I'll test the C2.

**Comment by Fourdee** [comment]
@swordfish6975 
Please can you send the pull request to Fourdee:testing (https://github.com/Fourdee/DietPi/tree/testing) branch. You may need to create a new pull request. Thanks.

**Pull request merged**